src/components/forms/JurisdictionSelect/index.tsx

```diff
@@ -53,13 +56,41 @@ const defaultProps: Partial<JurisdictionSelectProps> = {
  * This is simply a Higher Order Component that wraps around AsyncSelect
  */
 const JurisdictionSelect = (props: JurisdictionSelectProps & FieldProps) => {
-  const { apiEndpoint, cascadingSelect, field, form, labelFieldName, params, serviceClass } = props;
+  const {
+    apiEndpoint,
+    cascadingSelect,
+    field,
+    form,
+    labelFieldName,
+    params,
+    serviceClass,
+    interventionType,
+  } = props;
 
   const [parentId, setParentId] = useState<string>('');
   const [hierarchy, setHierarchy] = useState<SelectOption[]>([]);
   const [shouldMenuOpen, setShouldMenuOpen] = useState<boolean>(false);
   const [closeMenuOnSelect, setCloseMenuOnSelect] = useState<boolean>(false);
 
+  /** set state variables to default and clear form input */
+  const resetStateVars = () => {
+    // reset the state vars
+    setParentId('');
+    setHierarchy([]);
+    setShouldMenuOpen(false);
+    setCloseMenuOnSelect(false);
+    // set the Formik field value
+    if (form && field) {
+      form.setFieldValue(field.name, '');
+    }
+  };
+
+  useEffect(() => {
+    if (interventionType === InterventionType.MDAPoint) {
+      resetStateVars();
```

Comment:
@ciremusyoka why do we need to do this?

Reply:
@moshthepitt This is meant to make the code for resetting the component state and form more reusable. 
Needed to use the function [here](https://github.com/onaio/reveal-frontend/blob/814ad39c716c2990a0f8d1e69df3c0783db91b3a/src/components/forms/JurisdictionSelect/index.tsx#L88-L92) too.

Reply:
@ciremusyoka thanks.  However, why do we need to call `resetStateVars()` when the component mounts and `interventionType === InterventionType.MDAPoint`?

Reply:
@moshthepitt Will `IRS, FI and MDAPoint` intervention types use similar endpoint to pull locations?

Reply:
Yes